operations/helm/charts/alloy/values.yaml

```diff
@@ -120,6 +120,20 @@ image:
 rbac:
   # -- Whether to create RBAC resources for Alloy.
   create: true
+    # -- Grant rules to allow created RBAC resources to work with `discovery.kubernetes` component.
+  allowDiscoveryKubernetes: true
+  # -- Grant rules to allow created RBAC resources to work with `loki.source.kubernetes` and `loki.source.podlogs` component.
+  allowLokiSource: true
+  # -- Grant rules to allow created RBAC resources to work with `mimir.rules.kubernetes` component.
+  allowMimirRulesKubernetes: true
+  # -- Grant rules to allow created RBAC resources to work with `prometheus.kubernetes.*` components.
+  allowPrometheusKubernetes: true
+  # -- Grant rules to allow created RBAC resources to work with `eventhandler` integration.
+  allowEventHandler: true
+  # -- Grant rules to allow created RBAC resources to work with `remote.kubernetes.*` components.
+  allowRemoteKubernetes: true
+  # -- Grant rules to allow created RBAC resources to work with `otelcol.processor.k8sattributes` component.
+  allowOtelcolProcessorK8sAttributes: true
```

Comment:
I'm a little worried about tying Helm chart settings to knowledge of specific component names:

* The names might diverge with what's in Alloy over time (even right now, `eventhandler` from the comment is stale and from static mode; it should be `loki.source.kubernetes_events`). 
* There may be other components that require the same permissions, but aren't documented in the comments here. 

Can we find a way to partition these by permissions being granted rather than the main component that uses them? For example, instead of `allowLokiSource` it might be broken down into `readPods` and `readPodLogs`. 

Doing this is probably going to lead to having way more settings here, but it will also be decoupled from components and give users more granular control over what Alloy has permissions to access.